### ISSUE TITLE
Server::actix: is possible to inject app_data to be used with `#[oasgen] handler`

### ISSUE THREAD

**Issue by niklasad1** [kurtbuilds/oasgen#21]
Hey,

I'm playing around with this library and I would like to access some context/app data within my handler and I don't understand whether it possible or not (or I'm just doing something really stupid)

For example in actix-web it's possible to do:

```rust
pub async fn submissions(db: web::Data<Database>) -> Result<HttpResponse, Error> {
    let submissions = db.get_all_submissions().await.map_err(BoxError::from)?;
    let body = serde_json::to_string(&submissions).map_err(BoxError::from)?;

    Ok(HttpResponse::build(StatusCode::OK)
        .content_type("application/json; charset=utf-8")
        .body(body))
}

async fn main() {
         HttpServer::new(move || {
                App::new()
                    .app_data(web::Data::new(db.clone()))
                    .route("/submissions", web::get().to(submissions))
            })
            .bind(listen_addr)?
            .run()
            .await
}
```

I would like to something similar with oasgen such as:

```rust
#[oasgen]
// this doesn't work
pub async fn submissions(db: web::Data<Database>) -> Result<HttpResponse, Error> {
    let submissions = db.get_all_submissions().await.map_err(BoxError::from)?;
    let body = serde_json::to_string(&submissions).map_err(BoxError::from)?;

    Ok(HttpResponse::build(StatusCode::OK)
        .content_type("application/json; charset=utf-8")
        .body(body))
}

async fn main() {
    let server = oasgen::Server::actix()
        .route_json_spec("/docs/openapi.json")
        .route_yaml_spec("/docs/openapi.yaml")
        .swagger_ui("/docs/")
        .get("/submissions", submissions)
        .write_and_exit_if_env_var_set("./openapi.yaml")
        .freeze();

    HttpServer::new(move || {
        App::new()
            .app_data(db.clone())
            .service(data.clone().into_service())
    })
    .bind(listen_addr)?
    .run()
    .await
}
```

So, is it possible or not?

**Comment by kurtbuilds** [comment]
Can you share the error you're encountering?

I stopped using actix several years ago and switched to axum, so while there are tests for oasgen's actix functionality, it hasn't received as much love. Without seeing the error, my guess is there's a simple blank implementation missing for `web::Data`. I'd welcome a PR to fix it. If I understand correctly, actix's `web-data` is very similar to axum `State` - which can be used as inspiration for the PR.

**Comment by niklasad1** [comment]
```bash
thread '<unnamed>' panicked at ../.cargo/registry/src/index.crates.io-6f17d22bba15001f/oasgen-core-0.20.1/src/schema/actix.rs:18:9:
Call parameters() for Data, not schema().
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: routes::_::__INVENTORY::{{closure}}
   3: oasgen::server::actix::<impl oasgen::server::Server<oasgen::server::actix::ActixRouter>>::get
   4: <core::pin::Pin<P> as core::future::future::Future>::poll
   5: tokio::runtime::scheduler::current_thread::CoreGuard::block_on
   6: tokio::runtime::context::runtime::enter_runtime
   7: tokio::runtime::runtime::Runtime::block_on
```

**Comment by kurtbuilds** [comment]
Thank you. Fixed in 0.20.2.
